lib/install/install.go

```diff
@@ -180,9 +191,24 @@ func (i *Installer) startExecuteLoop() {
 		for {
 			select {
 			case req := <-i.execC:
-				if err := i.execute(req); err != nil {
-					i.sendError(err)
+				status, err := i.execute(req)
+				select {
+				case <-i.execDoneC:
+					// Empty the result channel
+				default:
+				}
+				if err != nil {
 					i.WithError(err).Warn("Failed to execute.")
```

Comment:
Nit: I realize this is existing code, but this should probably be updated to include what failed to execute as fields in the error...